src/main/java/com/faforever/client/fx/PlayerContextMenuController.java

```diff
@@ -0,0 +1,249 @@
+        addFoeItem.setVisible(socialStatus != FOE && socialStatus != SELF);
+        removeFoeItem.setVisible(socialStatus == FOE);
+        reportItem.setVisible(socialStatus != SELF);
+        joinGameItem.setVisible(socialStatus != SELF
+            && (playerStatus == PlayerStatus.LOBBYING || playerStatus == PlayerStatus.HOSTING)
+            && game != null && game.getGameType() != GameType.MATCHMAKER);
+        watchGameItem.setVisible(playerStatus == PlayerStatus.PLAYING);
+      });
+    };
+  }
+
+  public ContextMenu getContextMenu() {
+    return getRoot();
+  }
+
+  public void setPlayer(PlayerBean player) {
+    this.player = player;
+
+    setModeratorOptions(moderatorService.getPermissions(), player);
+
+    WeakInvalidationListener weakChatUserPropertyListener = new WeakInvalidationListener(playerPropertyInvalidationListener);
+    JavaFxUtil.addListener(player.socialStatusProperty(), weakChatUserPropertyListener);
+    JavaFxUtil.addAndTriggerListener(player.statusProperty(), weakChatUserPropertyListener);
+  }
+
+
+  public void onShowUserInfoSelected() {
+    PlayerBean player = getPlayer();
+    PlayerInfoWindowController playerInfoWindowController = uiService.loadFxml("theme/user_info_window.fxml");
+    playerInfoWindowController.setPlayer(player);
+    playerInfoWindowController.setOwnerWindow(getRoot().getOwnerWindow());
+    playerInfoWindowController.show();
+  }
+
+  public void onCopyUsernameSelected() {
+    PlayerBean player = getPlayer();
+    ClipboardUtil.copyToClipboard(player.getUsername());
+  }
+
+  public void onAddFriendSelected() {
+    PlayerBean player = getPlayer();
+    if (player.getSocialStatus() == FOE) {
+      playerService.removeFoe(player);
+    }
+    playerService.addFriend(player);
+  }
+
+  public void onRemoveFriendSelected() {
+    PlayerBean player = getPlayer();
+    playerService.removeFriend(player);
+  }
+
+  public void onAddFoeSelected() {
+    PlayerBean player = getPlayer();
+    if (player.getSocialStatus() == FRIEND) {
+      playerService.removeFriend(player);
+    }
+    playerService.addFoe(player);
+  }
+
+  public void onRemoveFoeSelected() {
+    PlayerBean player = getPlayer();
+    playerService.removeFoe(player);
+  }
+
+  public void onReport() {
+    ReportDialogController reportDialogController = uiService.loadFxml("theme/reporting/report_dialog.fxml");
+    PlayerBean player = getPlayer();
+    reportDialogController.setOffender(player);
+    reportDialogController.setOwnerWindow(getRoot().getOwnerWindow());
+    reportDialogController.show();
+  }
+
+  public void onJoinGameSelected() {
+    PlayerBean player = getPlayer();
+    joinGameHelper.join(player.getGame());
+  }
+
+  public void onWatchGameSelected() {
+    PlayerBean player = getPlayer();
+    try {
+      replayService.runLiveReplay(player.getGame().getId());
+    } catch (Exception e) {
+      log.error("Cannot display live replay", e.getCause());
```

Comment:
Are you sure all exceptions thrown here will have a cause

Reply:
I'm not. This was just copied over.